### ISSUE TITLE
[Bug] Default scan settings are not satisfying legacy devices (< SDK 29)

### ISSUE THREAD

**Issue by NoelChew** [LeandroSQ/android-ble-made-easy#16]
Hello, thank you for creating this awesome library! 

I am currently running into some issues on older devices. I tried running the sample project with 3 different Android versions: Android 6, Android 9 and Android 10. So far only the sample app installed on the Android 10 device managed to output scan results. I have also tested the Android 6 and Android 9 devices with other BLE apps to confirm that the BLE is not faulty. Has the library been tested with pre-Android10 devices before?

**Comment by LeandroSQ** [comment]
Thank you for your comment!

Initially this was made for Android 6, just later added support for 10+.

I have found that some devices struggle more to scan nearby, most of the time it can be related to the scan settings.


I think I have an older device running Android 6 somewhere, will charge it and give it a try.

In the meantime, could you try different [ScanSettings](https://developer.android.com/reference/android/bluetooth/le/ScanSettings)?
Using the sample app you can edit [this file](https://github.com/LeandroSQ/android-ble-made-easy/blob/9868fe95384d657bee521e98d62fbadeaa737d78/app/src/main/java/quevedo/soares/leandro/blemadeeasy/view/scandevices/ScanDevicesFragment.kt#L138)

```kotlin
scanAsync(settings = ScanSettings
                          .Builder()
                          // Try different settings here
                          .build())
```

The default scan settings can be seen [here](https://github.com/LeandroSQ/android-ble-made-easy/blob/9868fe95384d657bee521e98d62fbadeaa737d78/lib/src/main/java/quevedo/soares/leandro/blemadeeasy/BLE.kt#L75)

**Comment by LeandroSQ** [comment]
I tried on Android 6 and the current default scan settings are not satisfying legacy devices.
Made a few changes on a separated branch for better defaults on legacy devices (< SDK 29)

Would you be so kind to test the sample app on [this branch](https://github.com/LeandroSQ/android-ble-made-easy/tree/feature/legacy_scan_settings) and share the results? Thank you!

**Comment by NoelChew** [comment]
Tested on the following Android versions: 6, 9, 10, 11, 12.
All tests are showing expected output by displaying BLE scan results.

Thank you for the fast fix!

**Comment by LeandroSQ** [comment]
No worries :)

Closed with #20

**Comment by LeandroSQ** [comment]
Fixed on v1.8.0
